lib/pbench/test/unit/server/user_identity_management/test_user_token_management.py

```diff
@@ -0,0 +1,102 @@
+
+    @responses.activate
+    def test_get_token(self, server_config, keycloak_oidc):
+        token_endpoint = keycloak_oidc.TOKEN_ENDPOINT
+
+        json_response = {
+            "access_token": "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJDeXVpZDFYU2F3eEJSNlp2azdNOXZBUnI3R3pUWnE2QlpDQjNra2hGMHRVIn0",
+            "expires_in": 300,
+            "refresh_expires_in": 1800,
+            "refresh_token": "eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJlNGVlMzc2Ni1mNTVkL",
+            "token_type": "Bearer",
+            "id_token": "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwiPt_h5LI707x4JLFCBeUMaYSkPhQrmXz2QQZ5qpD60Yo7w",
+            "not-before-policy": 0,
+            "session_state": "a46aca36-78e3-4b2a-90b3-7bd46d5ff70d",
+            "scope": "openid profile email",
+        }
+        responses.add(
+            responses.POST,
+            token_endpoint,
+            status=200,
+            json=json_response,
+        )
+        token_response = keycloak_oidc.get_user_token(
+            username=self.USERNAME, password=self.PASSWORD
+        )
+        assert token_response["access_token"] == json_response["access_token"]
+
+    @responses.activate
+    def test_invalid_user_credential(self, server_config, keycloak_oidc):
+        token_endpoint = keycloak_oidc.TOKEN_ENDPOINT
+        json_response = {
+            "error": "invalid_grant",
+            "error_description": "Invalid user credentials",
+        }
+        responses.add(
+            responses.POST,
+            token_endpoint,
+            status=401,
+            json=json_response,
+        )
+        token_response = keycloak_oidc.get_user_token(
+            username=self.USERNAME, password="wrong_pass"
+        )
+        assert token_response["error"] == json_response["error"]
+
+    def test_token_introspect_offline(self, keycloak_oidc, keycloak_mock_token):
+        options = {"verify_signature": True, "verify_aud": True, "verify_exp": True}
+        response = keycloak_oidc.token_introspect_offline(
+            token=keycloak_mock_token,
+            key="some_secret",
+            algorithms=["HS256"],
+            audience="test_client",
+            options=options,
+        )
+        assert response == {
+            "iat": 1659476706,
+            "exp": 1685396687,
+            "sub": "12345",
+            "aud": "test_client",
+        }
+
+    def test_token_introspect_wrong_aud_claim(self, keycloak_oidc, keycloak_mock_token):
+        options = {"verify_signature": True, "verify_aud": True, "verify_exp": True}
+        with pytest.raises(InvalidAudienceError):
+            keycloak_oidc.token_introspect_offline(
+                token=keycloak_mock_token,
+                key="some_secret",
+                algorithms=["HS256"],
+                audience="wrong_client",
+                options=options,
+            )
+
+    @responses.activate
+    def test_userinfo(self, server_config, keycloak_oidc, keycloak_mock_token):
```

Comment:
Yeah, true since I am not doing any postprocessing on the data that gets returned from the `userinfo_endpoint` there is nothing to test for at the moment, and one of the reasons for that is I don't know what postprocessing we might want to do in future once we get this data. It's more like testing a `responses` package right now :). However, this acts as a placeholder if we were to add more functionality to our function so we can just add a test for the new functionality here.

Reply:
There's not "nothing to test for" -- I pointed out in my original comment what the CUT does and therefore what a unit test could check.

That said, I'm not enthusiastic about placeholder tests:  testing has a cost which is supposed to produce a benefit; if this test isn't producing a benefit, then it's not worth the cost...and, if that's the case, then we should remove it, one way or another (I'm generally opposed to committing commented-out code, but, if you _really_ don't want to delete it, that would be better than actually having to run it).  I suspect that, once this test really has a purpose (if that day should actually come), then it won't be hard to reconstruct it, and the result is likely to be substantially different from the present form, anyway, so I suggest removing it from this PR.